src/FffPolygonGenerator.cpp

```diff
@@ -769,12 +769,25 @@ bool FffPolygonGenerator::isEmptyLayer(SliceDataStorage& storage, const LayerInd
 
 void FffPolygonGenerator::removeEmptyFirstLayers(SliceDataStorage& storage, size_t& total_layers)
 {
+    const auto get_highest_z = [&storage](const LayerIndex& layer_idx)
+    {
+        coord_t highest_z = 0;
+        for (std::shared_ptr<SliceMeshStorage>& mesh_ptr : storage.meshes)
+        {
+            auto& mesh = *mesh_ptr;
+            highest_z = layer_idx >= mesh.layers.size() ? highest_z : std::max(highest_z, mesh.layers[layer_idx].printZ);
+        }
+        return highest_z;
+    };
```

Comment:
Could you avoid creating a lambda function for using it only once ?

Reply:
Yes, I can.